kuma/redirects/redirects.py

```diff
@@ -762,7 +775,604 @@ def locale_redirect(pattern, to, prepend_locale=True, **kwargs):
              permanent=False),
 ]
 
-redirectpatterns = scl3_redirectpatterns + [
+zone_redirectpatterns = [
+    # The redirects for the case when the locale is not specified
+    # for a zone. It must be handled here, since for the cases when
+    # LocaleMiddleware handles the 404 response and redirects to the
+    # proper locale, it fails because the path is considered invalid.
+    redirect(
+        r'^(?:docs/)?Add-ons(?:/?|(?P<sub_path>[/$].+))$',
+        get_sub_path_handler(u'/docs/Mozilla/Add-ons'),
+        permanent=False,
+        decorators=shared_cache_control_for_zones),
+    redirect(
+        r'^(?:docs/)?Firefox(?:/?|(?P<sub_path>[/$].+))$',
+        get_sub_path_handler(u'/docs/Mozilla/Firefox'),
+        permanent=False,
+        decorators=shared_cache_control_for_zones),
+    redirect(
+        r'^(?:docs/)?Apps(?:/?|(?P<sub_path>[/$].+))$',
+        get_sub_path_handler(u'/docs/Web/Apps'),
+        permanent=False,
+        decorators=shared_cache_control_for_zones),
+    redirect(
+        r'^Learn(?:/?|(?P<sub_path>[/$].+))$',
```

Comment:
Yes, this is handled nicely (locale and `docs` prefix added in one redirect step) by the [`mindtouch_to_kuma_redirect` view](https://github.com/mozilla/kuma/blob/d8e9c74957db00317d17f0e81e1892672351371a/kuma/wiki/views/legacy.py#L96). I debated whether we wanted to rely on `mindtouch_to_kuma_redirect`, since I was unsure of its future, but I'm happy to go either way here. I'll plan on removing it as you suggest.